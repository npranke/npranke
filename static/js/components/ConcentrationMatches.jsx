import PropTypes from 'prop-types'
import React from 'react'

export function ConcentrationMatches(props) {
    const matchedPictures = props.matches.map((pictureid) => {
        return (
            <React.Fragment key={ pictureid }>
                { props.pictures[`picture${pictureid}`].image }
            </React.Fragment>
        )
    })

    return (
        <div className="matches">
            <div className="matches-header">
                <div className="left-text">
                    <span className="matches-portion">
                        matches:&emsp;
                    </span>
                    <span className="matches-portion number">
                        {
                            props.matches.length < 10
                                ? `0${props.matches.length}`
                                : `${props.matches.length}`
                        }
                    </span>
                </div>
                <div className="right-text">
                    <span className="time-portion">
                        time:&emsp;
                    </span>
                    <span className="time-portion number">
                        { props.minutes }
                    </span>
                    <span className="time-portion">
                        &ensp;:&ensp;
                    </span>
                    <span className="time-portion number">
                        { props.seconds }
                    </span>
                    <span className="time-portion">
                        &ensp;:&ensp;
                    </span>
                    <span className="time-portion number">
                        { props.centiseconds }
                    </span>
                </div>
            </div>
            <div className="matches-pictures">
                { matchedPictures }
            </div>
        </div>
    )
}

ConcentrationMatches.defaultProps = {
    centiseconds: '00',
    matches: [],
    minutes: '00',
    seconds: '00',
}

ConcentrationMatches.propTypes = {
    centiseconds: PropTypes.string,
    matches: PropTypes.arrayOf(PropTypes.string),
    minutes: PropTypes.string,
    pictures: PropTypes.objectOf(
        PropTypes.shape({
            image: PropTypes.node.isRequired,
        }).isRequired,
    ).isRequired,
    seconds: PropTypes.string,
}

export default ConcentrationMatches
