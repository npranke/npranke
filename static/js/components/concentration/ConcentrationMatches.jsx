import PropTypes from 'prop-types'
import React from 'react'

import useTimeRunner from '@components/hooks/TimeRunner'

function ConcentrationMatches(props) {
    const matchedPictures = props.matches.map((pictureid) => {
        return (
            <React.Fragment key={ pictureid }>
                { props.pictures[`picture${pictureid}`].image }
            </React.Fragment>
        )
    })

    const { centiseconds, seconds, minutes } = useTimeRunner(
        props.isTimeRunning,
    )

    return (
        <div className="matches">
            <div className="matches-header">
                <div className="left-text">
                    <span className="matches-text">
                        <span className="matches-portion">
                            matches:&ensp;
                        </span>
                        <span className="matches-portion number">
                            {
                                props.matches.length < 10
                                    ? `0${props.matches.length}`
                                    : `${props.matches.length}`
                            }
                        </span>
                    </span>
                    <span className="turns-text">
                        <span className="turns-portion">
                            turns:&ensp;
                        </span>
                        <span className="turns-portion number">
                            {
                                props.turns < 10
                                    ? `0${props.turns}`
                                    : `${props.turns}`
                            }
                        </span>
                    </span>
                </div>
                <div className="right-text">
                    <span className="time-text">
                        <span className="time-portion">
                            time:&emsp;
                        </span>
                        <span className="time-portion number">
                            { minutes }
                        </span>
                        <span className="time-portion">
                            &ensp;:&ensp;
                        </span>
                        <span className="time-portion number">
                            { seconds }
                        </span>
                        <span className="time-portion">
                            &ensp;:&ensp;
                        </span>
                        <span className="time-portion number">
                            { centiseconds }
                        </span>
                    </span>
                </div>
            </div>
            <div className="matches-pictures">
                { matchedPictures }
            </div>
        </div>
    )
}

ConcentrationMatches.defaultProps = {
    isTimeRunning: false,
    matches: [],
    turns: 0,
}

ConcentrationMatches.propTypes = {
    isTimeRunning: PropTypes.bool,
    matches: PropTypes.arrayOf(PropTypes.string),
    pictures: PropTypes.objectOf(
        PropTypes.shape({
            image: PropTypes.node.isRequired,
        }).isRequired,
    ).isRequired,
    turns: PropTypes.number,
}

export default ConcentrationMatches
