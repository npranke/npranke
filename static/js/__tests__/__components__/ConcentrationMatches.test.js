import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import React from 'react'

import ConcentrationMatches from '../../components/ConcentrationMatches'

Enzyme.configure({ adapter: new Adapter() })

const pictures = Array.from({ length: 12 }, (value, integer) => {
    return `${integer}`
}).reduce((pictureAccumulator, pictureid) => {
    pictureAccumulator[`picture${pictureid}`] = {
        image: <a href="image-link"><img src="" alt="" /></a>,
    }

    return pictureAccumulator
}, {})

describe('ConcentrationMatches', () => {
    test('has matches-header section', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches pictures={ pictures } />,
        )

        expect(
            concentrationMatches.find('.matches-header').exists(),
        ).toBe(true)
    })

    test('has matches-pictures section', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches pictures={ pictures } />,
        )

        expect(
            concentrationMatches.find('.matches-pictures').exists(),
        ).toBe(true)
    })

    test('shows matches number when less than 10', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches
                pictures={ pictures }
                matches={ ['0', '4', '10'] }
            />,
        )

        expect(
            concentrationMatches.find('.matches-portion.number').text(),
        ).toEqual('03')
    })

    test('shows matches number when 10', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches
                pictures={ pictures }
                matches={
                    Array.from(
                        { length: 10 },
                        (value, integer) => { return `${integer}` },
                    )
                }
            />,
        )

        expect(
            concentrationMatches.find('.matches-portion.number').text(),
        ).toEqual('10')
    })

    test('shows matches number when more than 10', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches
                pictures={ pictures }
                matches={
                    Array.from(
                        { length: 12 },
                        (value, integer) => { return `${integer}` },
                    )
                }
            />,
        )

        expect(
            concentrationMatches.find('.matches-portion.number').text(),
        ).toEqual('12')
    })

    test('shows turns number when less than 10', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches
                pictures={ pictures }
                turns={ 6 }
            />,
        )

        expect(
            concentrationMatches.find('.turns-portion.number').text(),
        ).toEqual('06')
    })

    test('shows turns number when 10', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches
                pictures={ pictures }
                turns={ 10 }
            />,
        )

        expect(
            concentrationMatches.find('.turns-portion.number').text(),
        ).toEqual('10')
    })

    test('shows turns number when more than 10', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches
                pictures={ pictures }
                turns={ 23 }
            />,
        )

        expect(
            concentrationMatches.find('.turns-portion.number').text(),
        ).toEqual('23')
    })

    test('has time-portion sections', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches pictures={ pictures } />,
        )

        expect(
            concentrationMatches.find('.time-portion'),
        ).toHaveLength(6)
    })

    test('has time-portion number sections', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches pictures={ pictures } />,
        )

        expect(
            concentrationMatches.find('.time-portion.number'),
        ).toHaveLength(3)
    })

    test('matches-pictures has no pictures with no matches', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches pictures={ pictures } />,
        )

        expect(
            concentrationMatches.find('.matches-pictures').children(),
        ).toHaveLength(0)
    })

    test('matches-pictures has some pictures with some matches', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches
                pictures={ pictures }
                matches={ ['0', '4', '7', '10', '11'] }
            />,
        )

        expect(
            concentrationMatches.find('.matches-pictures').children(),
        ).toHaveLength(5)
    })

    test('matches-pictures has all pictures with all matches', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches
                pictures={ pictures }
                matches={
                    Array.from(
                        { length: 12 },
                        (value, integer) => { return `${integer}` },
                    )
                }
            />,
        )

        expect(
            concentrationMatches.find('.matches-pictures').children(),
        ).toHaveLength(12)
    })
})

describe('ConcentrationMatches snapshot', () => {
    test('matches snapshot without matches', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches pictures={ pictures } />,
        )

        expect(concentrationMatches).toMatchSnapshot()
    })

    test('matches snapshot with some matches', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches
                pictures={ pictures }
                matches={ ['0', '4', '7', '10'] }
            />,
        )

        expect(concentrationMatches).toMatchSnapshot()
    })

    test('matches snapshot with all matches', () => {
        const concentrationMatches = shallow(
            <ConcentrationMatches
                pictures={ pictures }
                matches={
                    Array.from(
                        { length: 12 },
                        (value, integer) => { return `${integer}` },
                    )
                }
            />,
        )

        expect(concentrationMatches).toMatchSnapshot()
    })
})
