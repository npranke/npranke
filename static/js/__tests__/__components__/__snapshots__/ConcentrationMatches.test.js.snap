// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConcentrationMatches snapshot matches snapshot with all matches 1`] = `
<div
  className="matches"
>
  <div
    className="matches-header"
  >
    <div
      className="left-text"
    >
      <span
        className="matches-portion"
      >
        matches: 
      </span>
      <span
        className="matches-portion number"
      >
        12
      </span>
    </div>
    <div
      className="right-text"
    >
      <span
        className="time-portion"
      >
        time: 
      </span>
      <span
        className="time-portion number"
      >
        00
      </span>
      <span
        className="time-portion"
      >
         : 
      </span>
      <span
        className="time-portion number"
      >
        00
      </span>
      <span
        className="time-portion"
      >
         : 
      </span>
      <span
        className="time-portion number"
      >
        00
      </span>
    </div>
  </div>
  <div
    className="matches-pictures"
  >
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
  </div>
</div>
`;

exports[`ConcentrationMatches snapshot matches snapshot with some matches 1`] = `
<div
  className="matches"
>
  <div
    className="matches-header"
  >
    <div
      className="left-text"
    >
      <span
        className="matches-portion"
      >
        matches: 
      </span>
      <span
        className="matches-portion number"
      >
        04
      </span>
    </div>
    <div
      className="right-text"
    >
      <span
        className="time-portion"
      >
        time: 
      </span>
      <span
        className="time-portion number"
      >
        00
      </span>
      <span
        className="time-portion"
      >
         : 
      </span>
      <span
        className="time-portion number"
      >
        00
      </span>
      <span
        className="time-portion"
      >
         : 
      </span>
      <span
        className="time-portion number"
      >
        00
      </span>
    </div>
  </div>
  <div
    className="matches-pictures"
  >
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
    <a
      href="image-link"
    >
      <img
        alt=""
        src=""
      />
    </a>
  </div>
</div>
`;

exports[`ConcentrationMatches snapshot matches snapshot without matches 1`] = `
<div
  className="matches"
>
  <div
    className="matches-header"
  >
    <div
      className="left-text"
    >
      <span
        className="matches-portion"
      >
        matches: 
      </span>
      <span
        className="matches-portion number"
      >
        00
      </span>
    </div>
    <div
      className="right-text"
    >
      <span
        className="time-portion"
      >
        time: 
      </span>
      <span
        className="time-portion number"
      >
        00
      </span>
      <span
        className="time-portion"
      >
         : 
      </span>
      <span
        className="time-portion number"
      >
        00
      </span>
      <span
        className="time-portion"
      >
         : 
      </span>
      <span
        className="time-portion number"
      >
        00
      </span>
    </div>
  </div>
  <div
    className="matches-pictures"
  />
</div>
`;
